File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/ReservationQrResponse.java
package com.smartsolar.mobile.data.remote.dto;

public class ReservationQrResponse {
    private String reservationId;
    private String qrPayload;
    private String issuedAtUtc;

    public ReservationQrResponse() { }

    public ReservationQrResponse(String reservationId, String qrPayload, String issuedAtUtc) {
        this.reservationId = reservationId;
        this.qrPayload = qrPayload;
        this.issuedAtUtc = issuedAtUtc;
    }

    public String getReservationId() { return reservationId; }
    public String getQrPayload() { return qrPayload; }
    public String getIssuedAtUtc() { return issuedAtUtc; }
}

File: src/SmartSolar.Application/DTOs/Reservations/ReservationQrResponse.cs
/*
 * File: ReservationQrResponse.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Returns opaque QR reference payload and issuance timestamp to an authorized Prosumer.
 * Note: Keep this header and add/update method-level comments as the code evolves.
 */

namespace SmartSolar.Application.DTOs.Reservations;

public sealed record ReservationQrResponse(
    string ReservationId,
    string QrPayload,
    DateTime IssuedAtUtc
);
